vllm/model_executor/models/qwen2_5_vl.py

```diff
@@ -211,6 +224,242 @@ def all_gather_interleave(local_tensor, hidden_size: int, tp_size: int):
+
+    embed_dim_reduction_factor = (vision_model.spatial_merge_size *
+                                  vision_model.spatial_merge_size)
+
+    # Find the max length across all ranks
+    max_len_per_rank = max(
+        grouped_pixel_values_len) // embed_dim_reduction_factor
+    local_grid_thw_list = [grid_thw_list[i] for i in image_idxs_local.tolist()]
+
+    # Run the vision model on the local pixel_values_local
+    if pixel_values_local.shape[0] > 0:
+        image_embeds_local = vision_model(pixel_values_local,
+                                          local_grid_thw_list)
+    else:
+        # Handle empty case
+        image_embeds_local = torch.empty((0, vision_model.out_hidden_size),
+                                         device=pixel_values.device,
+                                         dtype=pixel_values.dtype)
+
+    # Pad the output based on max_len_per_rank
+    current_len = image_embeds_local.shape[0]
+    if current_len < max_len_per_rank:
+        padding_size = max_len_per_rank - current_len
+        padding = torch.zeros((padding_size, image_embeds_local.shape[1]),
+                              device=image_embeds_local.device,
+                              dtype=image_embeds_local.dtype)
+        image_embeds_local_padded = torch.cat([image_embeds_local, padding],
+                                              dim=0)
+    else:
+        image_embeds_local_padded = image_embeds_local
+
+    # Do all_gather to collect embeddings from all ranks
+    gathered_embeds = tensor_model_parallel_all_gather(
+        image_embeds_local_padded, dim=0)
+
+    # Remove padding and reconstruct per-rank embeddings
+    rank_embeddings = []
+    for rank in range(tp_size):
+        start_idx = rank * max_len_per_rank
+        end_idx = start_idx + (grouped_pixel_values_len[rank] //
+                               embed_dim_reduction_factor)
+        rank_embeddings.append(gathered_embeds[start_idx:end_idx])
+
+    # Reconstruct embeddings in the shuffled order
+    shuffled_embeddings = []
+    current_idx = 0
+    for rank in range(tp_size):
+        count = gpu_sample_counts[rank]
+        if count > 0:
+            # Get images assigned to this rank in shuffled order
+            rank_images = image_to_tp_rank[current_idx:current_idx + count]
+            rank_embed = rank_embeddings[rank]
+            # Split rank embeddings back to individual images
+            embed_start = 0
+            for img_idx in rank_images:
+                img_patches = (patches_per_image[img_idx] //
+                               embed_dim_reduction_factor)
+                img_embed = rank_embed[embed_start:embed_start + img_patches]
+                shuffled_embeddings.append(img_embed)
+                embed_start += img_patches
+            current_idx += count
+
+    # Unshuffle to restore original order
+    original_order_embeddings = [None] * len(shuffled_embeddings)
+    for shuffled_idx, original_idx in enumerate(unshuffle_indices):
```

Comment:
```suggestion
    for original_idx, shuffled_idx, in enumerate(shuffle_indices):
```
I think no need to use `unshuffle_indices` here